src/script/ui/ContextMenu.tsx

```diff
@@ -112,43 +124,48 @@ const ContextMenu: React.FC<ContextMenuProps> = ({entries, defaultIdentifier = '
       window.removeEventListener('mousedown', onMouseDown);
       window.removeEventListener('resize', cleanUp);
     };
-  }, [mainDiv, selected]);
+  }, [mainElement, selected]);
 
   return (
-    <div className="ctx-menu" ref={setMainDiv} style={{maxHeight: window.innerHeight, ...style}}>
+    <ul className="ctx-menu" ref={setMainElement} style={{maxHeight: window.innerHeight, ...style}}>
       {entries.map((entry, index) =>
         entry.isSeparator ? (
-          <div key={`${index}`} className="ctx-menu__separator" />
+          <li key={`${index}`} className="ctx-menu__separator" />
         ) : (
-          <div
+          <li
             key={`${index}`}
-            {...(entry.isDisabled
-              ? undefined
-              : {
-                  onClick: event => {
-                    event.preventDefault();
-                    cleanUp();
-                    entry.click?.(event.nativeEvent);
-                  },
-                  onMouseEnter: () => {
-                    setSelected(undefined);
-                  },
-                })}
-            data-uie-name={entry.identifier || defaultIdentifier}
-            title={entry.title || entry.label}
             className={cx('ctx-menu__item', {
               'ctx-menu__item--checked': entry.isChecked,
               'ctx-menu__item--disabled': entry.isDisabled,
               selected: entry === selected,
             })}
           >
-            {entry.icon && <Icon name={entry.icon} className="ctx-menu__icon" />}
-            <span>{entry.label}</span>
-            {entry.isChecked && <Icon.Check className="ctx-menu__check" data-uie-name="ctx-menu-check" />}
-          </div>
+            <button
+              id={`${entry.label.split(' ').join('_').toLowerCase()}-button`}
```

Comment:
Since we generate those ids in 2 different places, let's have a helper function that create them

Reply:
good point - extracted to the helper function